### ISSUE TITLE
Checkbox: Accessibility Issues

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2078]
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/trimble-oss/modus-web-components/blob/main/CONTRIBUTING.md)

### Describe the issue

Checkboxes are currently not accessible at all via screen readers. The label should be linked to the checkbox input but they are not.

![image](https://github.com/trimble-oss/modus-web-components/assets/1212885/f79ada52-b93d-4dc7-9713-3be5157ef3e1)

### Reduced test cases

I used NVDA on Windows 11 and tabbed across this test page:
https://coliff.github.io/modus-web-components-example/#checkbox

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Microsoft Edge

### What is the issue regarding ?

@trimble-oss/modus-web-components

### What version of npm package are you using ?

v.0.13.0

### Priority

Medium

### What product/project are you using Modus Components for ?

Modus

### What is your team/division name ?

Modus

### Are you willing to contribute ?

Maybe

### Are you using Modus Web Components in production ?

https://modus-web-components.trimble.com/
